src/program/processor/new_order.rs

```diff
@@ -303,10 +292,17 @@ fn process_new_order<'a, 'info>(
         base_atoms_to_withdraw,
         base_atoms_to_deposit,
     ) = {
+        let clock = Clock::get()?;
+        let mut get_clock_fn = || (clock.slot, clock.unix_timestamp as u64);
```

Comment:
why does `get_clock_fn` need to be `mut`?

Reply:
why is it a function at all? can't we just pass in this tuple? does it ever change?

Reply:
This is quite a bit cleaner IMO. It needs to be mut because you don't know whether the function that you're passing in changes external state